controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -152,6 +154,9 @@ func NewClient(ctx context.Context, config ClientConfiguration) (*Client, error)
 		return nil, errors.Wrap(err, "unable to create etcd client")
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, config.CallTimeout)
+	defer cancel()
```

Comment:
```suggestion
	if config.CallTimeout > 0 {
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, config.CallTimeout)
		defer cancel()
	}
```
Should we only use the CallTimeout if it's > 0? Looks like that's how the DialTimeout is mostly used in the etcd client.

With CallTimeout == 0 everything would time out immediately

(I know **we** always have it set)

@vincepri WDYT?

Reply:
I'm ok with adding this. I omitted it in this PR as we'd always have a default from the flags. This would mainly be beneficial if the client gets initiated for other reasons outside of the current scope. Curious if it would make sense to have a default within the `NewClient` function then instead?

Reply:
A sensible default for CallTimeout sounds a good way to prevent dangling calls in a general sense, rather than being infinite like today, which isn't ideal. There should be some validation that enforces a minimum CallTimeout (e.g. 5s) though, in that case users should receive an error when creating the client.

Reply:
Sounds good to me

Ideally we use the same sensible default in the flags and here in the client

Reply:
Updated to include a DefaultCallTimeout. Also realized that I forgot to port over a part where the CallTimeout actually gets assigned to the Client struct, so that's now fixed too!

Reply:
So in the previous version we would have ended up with 0? :)

Reply:
Yeah, my bad. We run an internal fork and I didn't cherry pick all of it over correctly. Explains why the e2e tests were failing!